cmd/outputs_cloud.go

```diff
@@ -0,0 +1,174 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/guregu/null.v3"
+
+	"go.k6.io/k6/cloudapi"
+	"go.k6.io/k6/cmd/state"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/lib/consts"
+	"go.k6.io/k6/metrics"
+)
+
+const defaultTestName = "k6 test"
+
+// createCloudTest performs some test and Cloud configuration validations and if everything
+// looks good, then it creates a test run in the k6 Cloud, unless k6 is already running in the Cloud.
+// It is also responsible for filling the test run id on the test options, so it can be used later.
+// It returns the resulting Cloud configuration as a json.RawMessage, as expected by the Cloud output,
+// or an error if something goes wrong.
+func createCloudTest(gs *state.GlobalState, test *loadedAndConfiguredTest) error {
+	conf, warn, err := cloudapi.GetConsolidatedConfig(
+		test.derivedConfig.Collectors[builtinOutputCloud.String()],
+		gs.Env,
+		"", // Historically used for -o cloud=..., no longer used (deprecated).
+		test.derivedConfig.Options.Cloud,
+		test.derivedConfig.Options.External,
+	)
+	if err != nil {
+		return err
+	}
+
+	if warn != "" {
+		gs.Logger.Warn(warn)
+	}
+
+	// If this is true, then it means that this code is being executed in the k6 Cloud.
+	// Therefore, we don't need to continue with the test run creation,
+	// as we don't need to create any test run.
+	//
+	// Precisely, the identifier of the test run is conf.TestRunID.
+	if conf.TestRunID.Valid {
+		return nil
+	}
+
+	// If not, we continue with some validations and the creation of the test run.
```

Comment:
nit: from here onwards it seems like a thing that could be in cloudapi and be reused.

Reply:
Not really sure about how to do this refactor in a way that it remains clearer for the future, I wrote you directly to discuss it in detail.

Reply:
Lets leave this for another PR